src/Azure.Functions.Cli/Actions/LocalActions/CreateFunctionAction.cs

```diff
@@ -75,6 +76,17 @@ public async override Task RunAsync()
             }
 
             var workerRuntime = GlobalCoreToolsSettings.CurrentWorkerRuntimeOrNone;
+            if (!FileSystemHelpers.FileExists(Path.Combine(Environment.CurrentDirectory, "local.settings.json")))
+            {
+                // we're assuming "func init" has not been run
+                await InitApp(workerRuntime);
+
+                if (workerRuntime == WorkerRuntime.None)
+                {
+                    workerRuntime = WorkerRuntimeLanguageHelper.GetCurrentWorkerRuntimeLanguage(_secretsManager);
```

Comment:
I'm concerned that this is the correct way to update workerRuntime. My worry is that GlobalCoreToolsSettings will continue to return None for runtime unless we run Init again, but as far as I can see we no longer have access to the args variable. Do we need to run Init with blank args or is that a concern?

Reply:
so I think ideally what you want is if you detect that you need init, you can just call the init action.

How about creating the `InitAction` object in `ParseArgs()` calling parse args on the init object there, then only calling RunAsync here if needed? You might need a way to communicate back from init action to this action what the workerRuntime is. Does this make sense?

Reply:
Yeah I think that makes sense, and I made some changes that I think match up with what you're saying. I guess I was trying to make some guesses on the runtime just in case they passed in a language to skip the runtime selection screen if possible, but I think this is cleaner and more consistent (and I'm sure more error-resistant)